pkg/odp/segment_manager.go

```diff
@@ -37,11 +37,14 @@ type DefaultSegmentManager struct {
 }
 
 // NewSegmentManager creates and returns a new instance of DefaultSegmentManager.
-func NewSegmentManager(cacheSize int, cacheTimeoutInSecs int64, odpConfig Config, apiManager SegmentAPIManager) *DefaultSegmentManager {
+func NewSegmentManager(cache Cache, cacheSize int, cacheTimeoutInSecs int64, odpConfig Config, apiManager SegmentAPIManager) *DefaultSegmentManager {
 	segmentManager := DefaultSegmentManager{
 		odpConfig:         odpConfig,
 		segmentAPIManager: apiManager,
-		segmentsCache:     NewLRUCache(cacheSize, cacheTimeoutInSecs),
+		segmentsCache:     cache,
+	}
+	if segmentManager.segmentsCache == nil {
+		segmentManager.segmentsCache = NewLRUCache(cacheSize, cacheTimeoutInSecs)
 	}
```

Comment:
```suggestion
		segmentsCache:     cache ?? NewLRUCache(cacheSize, cacheTimeoutInSecs),
	}
```

Reply:
golang does not support ternary operators such as `??` or `?:`.